winmm_midi driver: Issue initializing instruments (SC-55, MT-32, GM)

# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

[Description of the bug]

long story short, im doing the testing trying to implement MIDI support for PX68K and  the linux midi driver has been working fine, but the windows midi driver cannot properly set instruments, and causes notes to be held longer than it should. This has been reported years ago.

### Expected behavior

play midi music!

### Actual behavior

just to repeat, the midi driver works in linux for the core's implementation. but the windows driver (winmm_midi.c) does not. 

In general midi/ SC-55 mode, there is noticeable lack of pitch bends, so notes does sound right. other than that, the notes are being held longer and in returns overlaps with all other notes being played.

when using with MT-32, that has a different issue since instruments are not getting initialized at all.

a quick workaround for this issue, was to manipulate winmm_midi_write_long_event() in winmm_midi.c to send long_midi_msg as single byte messages. this work as workaround but ofcourse this is incorrect as per implementation, as the buffer for sysex are suppose to be sent in blocks. my limited midi understanding limits me in finding actual solution, but just point out something is wrong, and probably is just within the midi driver implementation on windows. 

### Steps to reproduce the bug

Just setup your midi device(GM/SC-55/MT-32) using the libretro api. for this test Akumajou Dracula is a good example.  If used with MT-32, the notes should be clearly wrong from the start. In SC-55/GM mode, the loading music after the mt-32/cm55 selection screen should have a note bending part in the end of it but its not there. other than that if you cancel anything that is playing, some notes will linger and wont release. or just load the game,  You can also goto music mode and play LOAD BGM if you missed the part in the intro.
(btw, a PR in px68k has to be merged first or midi will remain pianos all the time regardless of midi types)

### Bisect Results

happens since commit

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit] latest master

### Environment information

- OS: [The operating system you're running] Windows 11
- Compiler: [In case you are running local builds] msys2 14.1.0

Related issue and some relevant discussion:
https://github.com/libretro/px68k-libretro/issues/164

just for comparison, i have crudely implemented midi handling directly in px68k using winmm (midiOutOpen, midiOutLongMsg, midiOutShortMsg etc) and this works fine with Windows GS Synth, MT-32, or SC-55 synths (munt and nuke). but like the libretro api to work and use it instead:
here is the repo if anyone wants to check or confirm:
https://github.com/negativeExponent/px68k-libretro/tree/midi_direct_using_winmm

Oh wow. Sysex produces error 11, which is MMSYSERR_INVALPARAM, invalid parameter, probably there is something wrong in the buffer setup. But the bend looks weird, this value is not in any reference:
`[ERROR] [MIDI]: midiStreamOut failed with error 68.`

according to midiOutGetErrorText, error codes means:

```
[ERROR] [MIDI]: midiStreamOut failed with error 11.
[ERROR] midiStreamOut error: An invalid parameter was passed to a system function.
```

```
[ERROR] [MIDI]: midiStreamOut failed with error 68.
[ERROR] midiStreamOut error: The current MIDI Mapper setup refers to a MIDI device that is not installed on the system.  Use MIDI Mapper to edit the setup.
```

--- a/midi/drivers/winmm_midi.c
+++ b/midi/drivers/winmm_midi.c
@@ -343,6 +343,9 @@ static bool winmm_midi_write_long_event(winmm_midi_buffer_t *buf,
 {
    DWORD i = buf->header.dwBytesRecorded / sizeof(DWORD);
 
+   /* data size has to be DWORD aligned */
+   data_size = (data_size + (sizeof(DWORD) - 1)) & ~(sizeof(DWORD) - 1);
+
    if (buf->header.dwBytesRecorded + sizeof(DWORD) * 3 + data_size >
          sizeof(DWORD) * WINMM_MIDI_BUF_LEN)
       return false;

